### PR TITLE
Feature: Agent connection types

### PR DESCRIPTION
**Pull request by bklang**
WIP

### CODE REVIEW

File: lib/electric_slide/call_queue.rb [hunk 1]
```diff
@@ -54,6 +65,14 @@ def get_agent(id)
     # Registers an agent to the queue
     # @param [Agent] agent The agent to be added to the queue
     def add_agent(agent)
+      case @connection_type
+      when :call
+        # FIXME: We want this to raise in the caller, and not kill the Queue actor
```
Comment (bklang) [inline lib/electric_slide/call_queue.rb]: @benlangfeld Do you know of a way to do this?

Comment (benlangfeld) [inline lib/electric_slide/call_queue.rb]: Use `#abort`.

Comment (bklang) [inline lib/electric_slide/call_queue.rb]: That works fine for this case (an actor that is already under supervision), but doesn't for the `#initialize` method:

```
Celluloid::Task::TerminatedError: task was terminated
	from /Users/bklang/.rvm/gems/ruby-2.1.5/gems/celluloid-0.15.2/lib/celluloid/tasks/task_fiber.rb:32:in `terminate'
	from /Users/bklang/.rvm/gems/ruby-2.1.5/gems/celluloid-0.15.2/lib/celluloid/actor.rb:404:in `block in cleanup'
	from /Users/bklang/.rvm/gems/ruby-2.1.5/gems/celluloid-0.15.2/lib/celluloid/actor.rb:404:in `each'
	from /Users/bklang/.rvm/gems/ruby-2.1.5/gems/celluloid-0.15.2/lib/celluloid/actor.rb:404:in `cleanup'
	from /Users/bklang/.rvm/gems/ruby-2.1.5/gems/celluloid-0.15.2/lib/celluloid/actor.rb:375:in `shutdown'
	from /Users/bklang/.rvm/gems/ruby-2.1.5/gems/celluloid-0.15.2/lib/celluloid/actor.rb:367:in `handle_crash'
	from /Users/bklang/.rvm/gems/ruby-2.1.5/gems/celluloid-0.15.2/lib/celluloid/actor.rb:187:in `rescue in run'
	from /Users/bklang/.rvm/gems/ruby-2.1.5/gems/celluloid-0.15.2/lib/celluloid/actor.rb:171:in `run'
	from /Users/bklang/.rvm/gems/ruby-2.1.5/gems/celluloid-0.15.2/lib/celluloid/actor.rb:157:in `block in initialize'
	from /Users/bklang/.rvm/gems/ruby-2.1.5/gems/celluloid-0.15.2/lib/celluloid/thread_handle.rb:13:in `block in initialize'
	from /Users/bklang/.rvm/gems/ruby-2.1.5/gems/celluloid-0.15.2/lib/celluloid/internal_pool.rb:100:in `call'
	from /Users/bklang/.rvm/gems/ruby-2.1.5/gems/celluloid-0.15.2/lib/celluloid/internal_pool.rb:100:in `block in create'
	from /Users/bklang/.rvm/gems/ruby-2.1.5/gems/logging-1.8.2/lib/logging/diagnostic_context.rb:323:in `call'
	from /Users/bklang/.rvm/gems/ruby-2.1.5/gems/logging-1.8.2/lib/logging/diagnostic_context.rb:323:in `block in create_with_logging_context'
	from (celluloid):0:in `remote procedure call'
	from /Users/bklang/.rvm/gems/ruby-2.1.5/gems/celluloid-0.15.2/lib/celluloid/calls.rb:92:in `value'
	from /Users/bklang/.rvm/gems/ruby-2.1.5/gems/celluloid-0.15.2/lib/celluloid/proxies/sync_proxy.rb:33:in `method_missing'
	from /Users/bklang/.rvm/gems/ruby-2.1.5/gems/celluloid-0.15.2/lib/celluloid/proxies/actor_proxy.rb:20:in `_send_'
	from /Users/bklang/.rvm/gems/ruby-2.1.5/gems/celluloid-0.15.2/lib/celluloid.rb:189:in `new'
	from /Users/bklang/.rvm/gems/ruby-2.1.5/gems/celluloid-0.15.2/lib/celluloid/supervisor.rb:16:in `supervise_as'
	from /Users/bklang/.rvm/gems/ruby-2.1.5/gems/celluloid-0.15.2/lib/celluloid.rb:214:in `supervise_as'
	from (irb):2
```

Comment (benlangfeld) [inline lib/electric_slide/call_queue.rb]: What semantics would you propose for a failure to initialise an actor that doesn't result in the actor failing to initialise?

Comment (bklang) [inline lib/electric_slide/call_queue.rb]: I would expect the strack trace to include a reference to the line of code that failed. In the above stack trace, the class failing to initialize is nowhere to be seen.

File: lib/electric_slide/agent.rb [hunk 2]
```diff
@@ -35,5 +35,14 @@ def dial_options_for(queue, queued_call)
     {}
   end
 
+  def join(queued_call)
+    # For use in queues that need bridge connections
+    @call.join queued_call
```
Comment (benlangfeld) [inline lib/electric_slide/agent.rb]: Where does `@call` come from here?

**Pull request merged**